src/gt4py/next/program_processors/runners/dace_common/dace_backend.py

```diff
@@ -88,10 +88,15 @@ def _get_shape_args(
     for name, value in args.items():
         for sym, size in zip(arrays[name].shape, value.shape, strict=True):
             if isinstance(sym, dace.symbol):
-                assert sym.name not in shape_args
-                shape_args[sym.name] = size
+                if sym.name not in shape_args:
+                    shape_args[sym.name] = size
+                elif shape_args[sym.name] != size:
+                    # TODO(edopao): This case is only hit if all fields in a tuple have the same dims and sizes.
```

Comment:
I am confused by the comment. Can you explain?

Reply:
You are right, I can write it better:

> The same shape symbol is used by all fields of a tuple, because the current assumption is that all fields in a tuple have the same dims and sizes. Therefore, this if-branch only exists to ensure that the array size (i.e. the value assigned to the shape symbol) is the same for all fields in a tuple.
> TODO(edopao): change to `assert sym.name not in shape_args` to ensure that shape symbols are unique, once the assumption on tuples is removed.